include/zephyr/net/net_context.h

```diff
@@ -604,6 +604,12 @@ static inline int net_context_get_can_filter_id(struct net_context *context)
  *
  * @return Network context IP protocol.
  */
+/* TODO: This method is now a misnomer - it returns the subprotocol of
```

Comment:
Theoretically, we should deprecate the old function name, and make it call the new renamed function, I think however in this particular case it's ok to just rename, `net_context()` isn't really intended to be used by applications so I guess we could consider it internal.

Reply:
Ok, I renamed the method then without deprecating it and mentioned the decision in the commit comment so it can be reverted if it causes trouble.